File: LeetCode_Python/1_twoSum.py
from typing import List


class TwoSum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsMap = {}

        for i in range(len(nums)):
            complement = target - nums[i]

            if complement in numsMap:
                return [numsMap[complement], i]

            numsMap[nums[i]] = i
            
if __name__ == "__main__":
    solution = TwoSum()
    nums = [2, 7, 11, 15]
    target = 9
    
    print(solution.twoSum(nums, target))
    
    

File: LeetCode_Java/1_twoSum.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class TwoSum {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> numsMap = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            int complement = target - nums[i];

            if(numsMap.containsKey(complement)){
                return new int[] {numsMap.get(complement), i};
            }

            numsMap.put(nums[i], i);
        }

        return new int[] {};
        
    }
    public static void main(String[] args) {
        TwoSum solution = new TwoSum();

        int[] nums = {2, 7, 11, 15};
        int target = 9;
        
        int[] result = solution.twoSum(nums, target);
        
        System.out.println(Arrays.toString(result));
    }
}
